adam/learner/subset.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict, Generic, Mapping, Set, Tuple
+
+from attr import Factory, attrib, attrs
+from immutablecollections import immutabledict
+
+from adam.language import LinguisticDescriptionT
+from adam.learner import LanguageLearner, LearningExample
+from adam.perception import PerceptionT, PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    PropertyPerception,
+    DevelopmentalPrimitivePerceptionFrame,
+)
+
+
+@attrs
+class SubsetLanguageLearner(
+    Generic[PerceptionT, LinguisticDescriptionT],
+    LanguageLearner[PerceptionT, LinguisticDescriptionT],
+):
+    """
+    An implementation of `LanguageLearner` for subset learning based approach for single object detection.
+    """
+
+    _descriptions_to_properties: Dict[
```

Comment:
Why not just map directly from the tokens to the properties without the `LinguisticDescriptionT` as middle-man?